Kernel/Syscalls/mmap.cpp

```diff
@@ -489,9 +490,42 @@ KResultOr<int> Process::sys$munmap(Userspace<void*> addr, size_t size)
         return 0;
     }
 
-    // FIXME: We should also support munmap() across multiple regions. (#175)
+    // Try again while checkin multiple regions at a time
+    // slow: without caching
+    const auto& regions = space().find_regions_intersecting(range_to_unmap);
+    // if there still no regions found error out
+    if (!regions.size())
+        return EINVAL;
 
-    return EINVAL;
+    // check if any of the regions is invalid in somesort, to not accientally
+    // crash with just half a region map left
```

Comment:
I don't understand this comment. Why would we "crash with just half a region map left" if there's a non-mmap region in the range?

It's correct to check for this, as we don't want `munmap` to unmap the signal trampoline page (which is non-mmap.)

Reply:
we check beforehand if any of the pages are not mmap, to avoid an error halfway through, at which point some pages would have been unmapped and others won't, or worse we split a range and haven't set it up correctly again.

If we now at which range the signal trampoline is wecould check with intersection and then use the lambda aproach as stated above

Reply:
This loop is fine, I just didn't understand the comment since it talks about crashing which is a specific thing in this context (`Process::crash()`) and it didn't seem to apply :)

Reply:
Is the signal trampoline the only not mmap region in a process?

Reply:
No, there is also the original executable/loader mappings, as well as the main thread stack.

Reply:
So explicitly checking for these wouldnt be woth it?

Do we even need the original executable/loader mappings at whis point?

Reply:
At least I've adjusted that wording now to some degree to
>  check if any of the regions is not mmaped, to not accientally error-out with just half a region map left